plugins/extractit/icon_extractit/util/extractor.py

```diff
@@ -12,11 +13,108 @@
 import zipfile
 import pdfplumber
 from pdfminer.pdfparser import PDFSyntaxError
+import openpyxl
 
 DEFAULT_ENCODING = "utf-8"
 
 
-def extract(provided_regex: str, provided_string: str, provided_file: str, keep_original_url: bool = False) -> list:
+def _get_cell_number_format(cell_number_format: str) -> str:
+    """The function reformats obtained cell number format formatted for datetime strftime() method and returns it
+
+    :param cell_number_format: Cell number format obtained from openpyxl
+    :type cell_number_format: str
+
+    :return: Returns formatted for datetime cell number format
+    :rtype: str
+    """
+
+    return (
+        cell_number_format.replace("mm", "%m")
+        .replace("dd", "%d")
+        .replace("yyyy", "%Y")
+        .replace("yy", "%y")
+        .replace('"', "")
+        .lstrip("0")
+    )
+
+
+def _get_non_empty_cells(input_unzipped_file: zipfile.ZipFile) -> List[str]:
+    """Gets all the cell names (i.e. A1) when it's value is not empty
+
+    :param input_unzipped_file: Unzipped file containing all the xml files
+    :type input_unzipped_file: zipfile.ZipFile
+
+    :return: Returns list containing non-empty cell names
+    :rtype: List[str]
+    """
+
+    output_cell_values = []
+    for file_ in input_unzipped_file.infolist():
+        if "worksheet" in file_.filename and file_.filename.endswith("xml"):
+            output_cell_values += regex.findall(
+                r'r="[A-Z]+\d+"', input_unzipped_file.read(file_).decode(DEFAULT_ENCODING)
+            )
+    return list({cell[3:].strip('"') for cell in output_cell_values})
+
+
+def _extract_dates_from_cells(input_sheet: str, cells_with_values: List[str]) -> List[str]:
+    """Extracts the dates from all non empty cells that are formatted as Date and Time cells
+    with the date format defined in the xlsx sheet
+
+    :param input_sheet: Name of the sheet from where the data will be obtained
+    :type input_sheet: str
+    :param cells_with_values: List that contains all the non-empty cell names
+    :type cells_with_values: List[str]
+
+    :return: Returns list containing Date Time values from the input cells
+    :rtype: List[str]
+    """
+
+    output_dates = []
+    for cell in cells_with_values:
+        cell_value = input_sheet[cell].value
```

Comment:
It depends on the type. If it's a plain text in the xlsx file - then it will be a string. If it's a number with a specific format - then it will be int or float. Same for datetime 😄